src/test/java/org/springframework/data/redis/cache/RedisCacheUnitTests.java

```diff
@@ -46,76 +37,21 @@ void cacheRetrieveValueCallsCacheWriterRetrieveCorrectly() throws Exception {
 
 		RedisCacheWriter mockCacheWriter = mock(RedisCacheWriter.class);
 
-		doReturn(CompletableFuture.completedFuture("TEST".getBytes()))
-				.when(mockCacheWriter).retrieve(anyString(), any(byte[].class));
+		when(mockCacheWriter.supportsAsyncRetrieve()).thenReturn(true);
+		when(mockCacheWriter.retrieve(anyString(), any(byte[].class)))
+				.thenReturn(CompletableFuture.completedFuture("TEST".getBytes()));
 
 		RedisCache cache = new RedisCache("TestCache", mockCacheWriter,
-				RedisCacheConfiguration.defaultCacheConfig());
+				RedisCacheConfiguration.defaultCacheConfig().serializeValuesWith(SerializationPair.byteArray()));
 
-		CompletableFuture<byte[]> value = cache.retrieveValue("TestKey");
+		CompletableFuture<byte[]> value = (CompletableFuture<byte[]>) cache.retrieve("TestKey");
 
 		assertThat(value).isNotNull();
 		assertThat(new String(value.get())).isEqualTo("TEST");
 
 		verify(mockCacheWriter, times(1)).retrieve(eq("TestCache"), isA(byte[].class));
+		verify(mockCacheWriter).supportsAsyncRetrieve();
 		verifyNoMoreInteractions(mockCacheWriter);
 	}
 
-	@Test // GH-2650
```

Comment:
What happened to these test case methods?

Reply:
These are testing internals (to some extent similar to testing getters and setters) and we have already tests for storing null values and null value caching in `RedisCacheTests` so it doesn't make sense to have redundancy causing additional maintenance efforts.

Reply:
`RedisCacheTests` are "Integration" Tests whereas Unit Testing is a form of White-box testing. These tests are ensuring the correct handling of `null` values regardless of infrastructure concerns and are not equivalent to getter and setter testing.

I agree with removing redundancy.